docs/core/resilience/http-resilience.md

```diff
@@ -0,0 +1,321 @@
+
+To configure a named resilience handler, call the `AddResilienceHandler` extension method with the name of the handler. The following example configures a named resilience handler called `"CustomPipeline"`.
+
+```csharp
+// The builder is of type, IHttpClientBuilder.
+builder.AddResilienceHandler(
+    "CustomPipeline",
+    static (ResiliencePipelineBuilder<HttpResponseMessage> builder,
+            ResilienceHandlerContext context) =>
+{
+    // See: https://www.pollydocs.org/strategies/retry.html
+    builder.AddRetry(new HttpRetryStrategyOptions
+    {
+        BackoffType = DelayBackoffType.Exponential,
+        MaxRetryAttempts = 5,
+        UseJitter = true
+    });
+
+    // See: https://www.pollydocs.org/strategies/circuit-breaker.html
+    builder.AddCircuitBreaker(new HttpCircuitBreakerStrategyOptions
+    {
+        SamplingDuration = TimeSpan.FromSeconds(10),
+        FailureRatio = 0.2,
+        MinimumThroughput = 3,
+        ShouldHandle = static args =>
+        {
+            // Handles HTTP status codes 408 and 429.
+            return ValueTask.FromResult(args is
+            {
+                Outcome.Result.StatusCode:
+                    HttpStatusCode.RequestTimeout or HttpStatusCode.TooManyRequests
+            });
+        }
+    });
+
+    // See: https://www.pollydocs.org/strategies/timeout.html
+    builder.AddTimeout(TimeSpan.FromSeconds(5));
+});
+```
+
+The preceding code:
+
+- Adds a resilience handler with the name `"CustomPipeline"` as the `pipelineName` to the service container.
+- Adds a retry strategy with exponential backoff, five retries, and jitter preference to the resilience builder.
+- Adds a circuit breaker strategy with a sampling duration of 10 seconds, a failure ratio of 0.2 (20%), a minimum throughput of three, and a predicate that handles `RequestTimeout` and `TooManyRequests` HTTP status codes to the resilience builder.
+- Adds a timeout strategy with a timeout of five seconds to the resilience builder.
+
+There are many options available for each of the resilience strategies. For more information, see the [Polly docs: Strategies](https://www.pollydocs.org/strategies).
+
+### Dynamic reload
+
+Polly supports dynamic reload of the resilience strategies. This means that you can change the configuration of the resilience strategies at runtime. To enable dynamic reload, use the appropriate `AddResilienceHandler` overload that exposes the `ResilienceHandlerContext`. Given the context, call `EnableReloads` of the corresponding resilience strategy options:
+
+```csharp
+builder.AddResilienceHandler(
+    "AdvancedPipeline",
+    static (ResiliencePipelineBuilder<HttpResponseMessage> builder,
+            ResilienceHandlerContext context) =>
+{
+    // Enable the reloads whenever the named options change
+    context.EnableReloads<RetryStrategyOptions>("my-retry-options");
+
+    // Retrieve the named options
+    var retryOptions =
```

Comment:
nit: wait for convenience API and use it here?

https://github.com/dotnet/extensions/pull/4497